crates/typst-library/src/model/outline.rs

```diff
@@ -365,136 +343,234 @@ cast! {
+        NonZeroUsize::ONE
+    }
+
+    /// Constructs the default prefix given the formatted numbering.
+    fn prefix(&self, numbers: Content) -> Content;
+
+    /// The body of the entry.
+    fn body(&self) -> Content;
 }
 
-/// Represents each entry line in an outline, including the reference to the
-/// outlined element, its page number, and the filler content between both.
-///
-/// This element is intended for use with show rules to control the appearance
-/// of outlines. To customize an entry's line, you can build it from scratch by
-/// accessing the `level`, `element`, `body`, `fill` and `page` fields on the
-/// entry.
-///
-/// ```example
-/// #set heading(numbering: "1.")
-///
-/// #show outline.entry.where(
-///   level: 1
-/// ): it => {
-///   v(12pt, weak: true)
-///   strong(it)
-/// }
+/// Represents an entry line in an outline.
 ///
-/// #outline(indent: auto)
-///
-/// = Introduction
-/// = Background
-/// == History
-/// == State of the Art
-/// = Analysis
-/// == Setup
-/// ```
-#[elem(name = "entry", title = "Outline Entry", Show)]
+/// With show-set and show rules on outline entries, you can richly customize
+/// the outline's appearance. See the
+/// [section on styling the outline]($outline/#styling-the-outline) for details.
+#[elem(scope, name = "entry", title = "Outline Entry", Show)]
 pub struct OutlineEntry {
     /// The nesting level of this outline entry. Starts at `{1}` for top-level
     /// entries.
     #[required]
     pub level: NonZeroUsize,
 
     /// The element this entry refers to. Its location will be available
-    /// through the [`location`]($content.location) method on content
+    /// through the [`location`]($content.location) method on the content
     /// and can be [linked]($link) to.
     #[required]
     pub element: Content,
 
-    /// The content which is displayed in place of the referred element at its
-    /// entry in the outline. For a heading, this would be its number followed
-    /// by the heading's title, for example.
-    #[required]
-    pub body: Content,
-
-    /// The content used to fill the space between the element's outline and
-    /// its page number, as defined by the outline element this entry is
-    /// located in. When `{none}`, empty space is inserted in that gap instead.
+    /// Content to fill the space between the title and the page number. Can be
+    /// set to `{none}` to disable filling.
     ///
     /// Note that, when using show rules to override outline entries, it is
     /// recommended to wrap the filling content in a [`box`] with fractional
     /// width. For example, `{box(width: 1fr, repeat[-])}` would show precisely
     /// as many `-` characters as necessary to fill a particular gap.
-    #[required]
+    ///
+    /// ```example
+    /// #set outline.entry(fill: line(length: 100%))
+    /// #outline()
+    ///
+    /// = A New Beginning
+    /// ```
+    #[borrowed]
+    #[default(Some(RepeatElem::new(TextElem::packed(".")).pack()))]
```

Comment:
```suggestion
    #[default(Some(RepeatElem::new(TextElem::packed("\u{2024}")).pack()))]
```

If a textual dot is used for the filler, it should be U+2024 ․ ONE DOT LEADER. Using a non-textual element would be better.

Quoting the [Unicode 16.0.0 Core Specification](https://www.unicode.org/versions/Unicode16.0.0/core-spec/chapter-6/#G13727):

> Usually leader dots are generated automatically by page formatting software and do not require the use of encoded characters. However, there are occasional plain text contexts in which a string of leader dots is represented as a sequence of characters. U+2024 ONE DOT LEADER and U+2025 TWO DOT LEADER are intended for such usage.

Refer to the full paragraph linked above for more information.

Reply:
Note that most fonts available in the web app don't have a glyph for that character

Reply:
Then, it may be better to use non-textual elements, or make the text non-selectable.

Reply:
I'm not sure that paragraph is saying _anything_ about what typesetting systems _should_ do when generating leaders. It reads more like instructions exclusively for plain text contexts to me.

I'm not sure how non-textual elements can be better here. The font's full stop has the correct size, is baseline-aligned, etc.

As for non-selectability: Yes, that would be good, but we don't currently have infrastructure for that.

Reply:
I think this sentence is also quite relevant: "These leader dot characters can be used to control, to a certain extent, the spacing of leader dots based on font design, in contexts where a simple sequence of full stops will not suffice."

This doesn't really apply to us since `repeat` has a gap property. I added a default gap now for a better out-of-the-box look.

Reply:
I'll keep the normal full stop for now.